pygeoif/functions.py

```diff
@@ -188,11 +190,60 @@ def compare_geo_interface(
         return False
 
 
+def move_coordinate(
+    coordinate: Sequence[float],
+    move_by: Sequence[float],
+    z: float = 0,
+) -> Tuple[float, ...]:
+    """
+    Move the coordinate by the given vector.
+
+    This forcefully changes the dimensions of the coordinate to match the latter.
+    >>> move_coordinate((0, 0), (-1, 1))
+    (-1, 1)
+    >>> move_coordinate((0, 0, 0), (-1, 1))
+    (-1, 1)
+    >>> move_coordinate((0, 0), (-1, 1, 0))
+    (-1, 1, 0)
+    """
+    if len(coordinate) > len(move_by):
+        return tuple(c + m for c, m in zip(coordinate, move_by))
+    return tuple(c + m for c, m in zip_longest(coordinate, move_by, fillvalue=z))
+
+
+def move_coordinates(
+    coordinates: Sequence[Any],
+    move_by: Sequence[float],
+    z: float = 0,
+) -> Sequence[Any]:
+    """
+    Move the coordinates recursively by the given vector.
+
+    This forcefully changes the dimension of each of the coordinate to match
+    the dimension of the vector.
+    >>> move_coordinates(((0, 0), (-1, 1)), (-1, 1))
+    ((-1, 1), (-2, 2))
+    >>> move_coordinates(((0, 0, 0), (-1, 1, 0)), (-1, 1))
+    ((-1, 1), (-2, 2))
+    >>> move_coordinates(((0, 0), (-1, 1)), (-1, 1, 0))
+    ((-1, 1, 0), (-2, 2, 0))
+    """
+    if isinstance(coordinates, (tuple, list)) and isinstance(
```

Comment:
I preferred to check types explicitly rather what catch TypeError exception.

Reply:
Thanks for pointing this out!
As far as I see, you refactored this lines to use `CoordinatesType`, not `Sequence[Any]`, so this function just can't accept generators anyway. 
Not sure if generators still need to be supported in `move_coordinates`?

Reply:
You still CAN pass in a generator, it is just that mypy will complain that you should not do that

Reply:
Thanks again for your help, your work will be in the next release (1.2) :+1:

Reply:
> You still CAN pass in a generator, it is just that mypy will complain that you should not do that

Yes, yes, you are right. I'm probably just too used to strict typing :)
Please consider PR #188 regarding it.